### ISSUE TITLE
Nested Picture element

### ISSUE THREAD

**Issue by ClarkeFL** [wp-media/imagify-plugin#570]
Hi,

There seems to be a bug when using imagify and [livecanvas.](https://livecanvas.com/) When using the picture tags to rewrite images and serve WEBP, it seems to nest the picture element within itself.

Here is an example:
`<a href="#">
	<picture>
		<source type="image/webp" srcset="example.png.png.webp">
		<picture>
			<source type="image/webp" srcset="example.png.png.webp">
			<picture>
				<source type="image/webp" srcset="example.png.png.webp">
				<picture>
					<source type="image/webp" srcset="example.png.png.webp">
					<picture>
						<source type="image/webp" srcset="example.png.png.webp">
						<picture>
							<source type="image/webp" srcset="example.png.webp">
							<picture>
								<source type="image/webp" srcset="example.png.png.webp">
								<img src="example.png" alt="">
							</picture>
						</picture>
					</picture>
				</picture>
			</picture>
		</picture>
	</picture>
</a>`

Im currently using a cdn with my site, else I would have used the rewrite rule. I belive this is a potential bug. It should not keep nesting itself.

**Comment by remyperona** [comment]
Hi @ClarkeFL,

Thank you for opening the issue!

Can you provide a sample of the original image? That will help us debug it.

**Comment by ClarkeFL** [comment]
Hi @Tabrisrp 

Here is a screenshot from our staging site:

<img width="1926" alt="Screen Shot 2021-06-08 at 8 10 39 am" src="https://user-images.githubusercontent.com/12570826/121094043-4e725300-c831-11eb-9abd-27142c94f7e8.png">

Hope this will help you debug it.

**Comment by remyperona** [comment]
@ClarkeFL Sorry what I meant was the code sample of the image before being converted to the picture element, not the final result. I should have been more clear on that.

**Comment by ClarkeFL** [comment]
@Tabrisrp 

All good, this is what it was originally

<img width="1425" alt="Screen Shot 2021-06-08 at 8 18 02 am" src="https://user-images.githubusercontent.com/12570826/121094536-26372400-c832-11eb-886c-ac4460c3b94a.png">

**Comment by remyperona** [comment]
Awesome thank you, we will investigate this 👍🏼

**Comment by mostafa-hisham** [comment]
I am not 100% sure but I think it is like this [issue #537](https://github.com/wp-media/imagify-plugin/issues/537)
@iCaspar can you please check this?
I already [groomed](https://github.com/wp-media/imagify-plugin/issues/537#issuecomment-876317282) #537

**Comment by remyperona** [comment]
@piotrbak Can you check this issue and see if the 537 fix also fixed this?

**Comment by iCaspar** [comment]
#537 was resolved on #590 which is now merged, so the fix should be in on `develop` at this point.

**Comment by piotrbak** [comment]
@Tabrisrp I wasn't able to reproduce that kind of problem using provided markup. On the mentioned site there are lots of nested `img` inside... We would either need access to the website or check if we're able to reproduce this using this premium builder... What do you think?

@iCaspar We were fixing a problem when there was a `<picture>` tag on the website already, from the screenshot provided by @ClarkeFL it looks like it's just a regular image...

**Comment by ClarkeFL** [comment]
@piotrbak You are correct. It is just a regular image. However I use the Imagify plugin to generate/serve WEBP images and then nest them in the ```<picture>``` tag. Therefore the ```<picture>``` tag is added by Imagify, but not only once. It keeps replicating it.

I unfortunately don't work for the company where I was a web developer. So I haven't used the plugin in a while and don't know if the issue has been fixed per say via issue #537 

I was using a plugin called [Live Canvas](https://livecanvas.com/) which allowed me to use bootstrap within WordPress. Also allowed me to gain access to the actual code of the pages I was creating. Within the code editor that where the issue occurred. Each time I saved the paged and refreshed it, the loop would occur (random amounts of the loop). 

I can do a test and get back to you guys to see if the issue has been fixed. I could also then give you access to the demo site if you need to view anything.

**Comment by Miraeld** [comment]
The recursive nesting of `<picture>` elements was fixed in PR #590 (commit 72703378), which added `remove_picture_tags()` in the Picture display class. Before inserting new picture markup, Imagify now strips any picture blocks already present in the content, so images that were already wrapped are not wrapped again on each save. That code is on develop in `classes/Picture/Display.php`, so this is resolved. Closing.
